--- gecko-sharp-0.6/gtkmozembed/Makefile.am.orig	2005-12-03 13:23:39.000000000 +0100
+++ gecko-sharp-0.6/gtkmozembed/Makefile.am	2005-12-03 13:29:17.000000000 +0100
@@ -45,11 +45,11 @@
 	$(build_sources) generated/*.cs -o gecko-sharp.dll
 
 install-data-local:
-	echo "$(GACUTIL) /i $(ASSEMBLY) /f /package gecko-sharp /root $(DESTDIR)$(libdir)";  \
-        $(GACUTIL) /i $(ASSEMBLY) /f /package gecko-sharp /root $(DESTDIR)$(libdir) || exit 1;
+	echo "$(GACUTIL) /i $(ASSEMBLY) /f /package gecko-sharp /root $(DESTDIR)$(prefix)/lib";  \
+        $(GACUTIL) /i $(ASSEMBLY) /f /package gecko-sharp /root $(DESTDIR)$(prefix)/lib || exit 1;
                                                                                 
 uninstall-local:
-	echo "$(GACUTIL) /u $(ASSEMBLY_NAME) /package gecko-sharp /root $(DESTDIR)$(libdir)"; \
-        $(GACUTIL) /u $(ASSEMBLY_NAME) /package gecko-sharp /root $(DESTDIR)$(libdir) || exit 1;
+	echo "$(GACUTIL) /u $(ASSEMBLY_NAME) /package gecko-sharp /root $(DESTDIR)$(prefix)/lib"; \
+        $(GACUTIL) /u $(ASSEMBLY_NAME) /package gecko-sharp /root $(DESTDIR)$(prefix)/lib || exit 1;
  
 EXTRA_DIST = $(API) $(RAW_API) $(sources) $(customs) $(METADATA) AssemblyInfo.cs.in gecko-sharp.dll.config
